modules/rostests/apitests/ntdll/NtDuplicateToken.c

```diff
@@ -98,30 +106,40 @@ DuplicateTokenAsEffective(VOID)
      * the token effective, with this potential privilege being
      * disabled by ourselves.
      */
-    DisablePrivilege(TokenHandle, L"SeImpersonatePrivilege");
+    if (!DisablePrivilege(TokenHandle, L"SeImpersonatePrivilege"))
+    {
+        skip("DisablePrivilege() failed\n");
+        goto cleanup;
+    }
 
     /* Query the total size of the token statistics structure */
     Status = NtQueryInformationToken(TokenHandle, TokenStatistics, NULL, 0, &Size);
-    if (!NT_SUCCESS(Status) && Status != STATUS_BUFFER_TOO_SMALL)
+    ok_ntstatus(Status, STATUS_BUFFER_TOO_SMALL);
+    if (Status != STATUS_BUFFER_TOO_SMALL)
     {
-        skip("Failed to query the total size for token statistics structure! (Status -> 0x%lx)\n", Status);
-        return;
+        if (NT_SUCCESS(Status))
+            skip("NtQueryInformationToken() succeeded unexpectedly\n");
+        else
+            skip("Failed to query the total size for token statistics structure\n");
+        goto cleanup;
     }
 
     /* Total size queried, time to allocate our buffer based on that size */
     TokenStats = RtlAllocateHeap(RtlGetProcessHeap(), 0, Size);
+    ok(TokenStats != NULL, "Failed to allocate our token statistics buffer\n");
     if (TokenStats == NULL)
     {
-        skip("Failed to allocate our token statistics buffer!\n");
-        return;
+        skip("TokenStats is NULL\n");
+        goto cleanup;
```

Comment:
You're already printing out that token statistics buffer hasn't been allocated due to a failure, no need for a `skip()` call to print the nearly same information again.